protocol-impl/src/main/java/io/zeebe/protocol/impl/encoding/BrokerInfo.java

```diff
@@ -209,22 +227,40 @@ public void wrap(final DirectBuffer buffer, int offset, final int length) {
       addressesDecoder.getAddress(addressBytes, 0, addressLength);
 
       this.addAddress(new UnsafeBuffer(apiNameBytes), new UnsafeBuffer(addressBytes));
+      offset +=
+          AddressesDecoder.apiNameHeaderLength()
+              + apiNameLength
+              + AddressesDecoder.addressHeaderLength()
+              + addressLength;
     }
 
+    offset += PartitionRolesDecoder.sbeHeaderSize();
     final PartitionRolesDecoder partitionRolesDecoder = bodyDecoder.partitionRoles();
     while (partitionRolesDecoder.hasNext()) {
       partitionRolesDecoder.next();
       this.addPartitionRole(partitionRolesDecoder.partitionId(), partitionRolesDecoder.role());
+      offset += 4 + 1;
     }
 
+    offset += PartitionLeaderTermsDecoder.sbeHeaderSize();
     final PartitionLeaderTermsDecoder partitionLeaderTermsDecoder =
         bodyDecoder.partitionLeaderTerms();
     while (partitionLeaderTermsDecoder.hasNext()) {
       partitionLeaderTermsDecoder.next();
       this.partitionLeaderTerms.put(
           partitionLeaderTermsDecoder.partitionId(), partitionLeaderTermsDecoder.term());
+      offset += 4 + 8;
     }
 
+    final int versionLength = bodyDecoder.versionLength();
```

Comment:
I think with `bodyDecoder.wrapVersion()` we don't need to keep track of the offset. The whole method can be replaced by:

```java
@Override
  public void wrap(final DirectBuffer buffer, int offset, final int length) {
    reset();

    final int frameEnd = offset + length;

    headerDecoder.wrap(buffer, offset);
    bodyDecoder.wrap(
        buffer,
        offset + headerDecoder.encodedLength(),
        headerDecoder.blockLength(),
        headerDecoder.version());

    nodeId = bodyDecoder.nodeId();
    partitionsCount = bodyDecoder.partitionsCount();
    clusterSize = bodyDecoder.clusterSize();
    replicationFactor = bodyDecoder.replicationFactor();

    final AddressesDecoder addressesDecoder = bodyDecoder.addresses();
    while (addressesDecoder.hasNext()) {
      addressesDecoder.next();
      final int apiNameLength = addressesDecoder.apiNameLength();
      final byte[] apiNameBytes = new byte[apiNameLength];
      addressesDecoder.getApiName(apiNameBytes, 0, apiNameLength);

      final int addressLength = addressesDecoder.addressLength();
      final byte[] addressBytes = new byte[addressLength];
      addressesDecoder.getAddress(addressBytes, 0, addressLength);

      this.addAddress(new UnsafeBuffer(apiNameBytes), new UnsafeBuffer(addressBytes));
    }

    final PartitionRolesDecoder partitionRolesDecoder = bodyDecoder.partitionRoles();
    while (partitionRolesDecoder.hasNext()) {
      partitionRolesDecoder.next();
      this.addPartitionRole(partitionRolesDecoder.partitionId(), partitionRolesDecoder.role());
    }

    final PartitionLeaderTermsDecoder partitionLeaderTermsDecoder =
        bodyDecoder.partitionLeaderTerms();
    while (partitionLeaderTermsDecoder.hasNext()) {
      partitionLeaderTermsDecoder.next();
      this.partitionLeaderTerms.put(
          partitionLeaderTermsDecoder.partitionId(), partitionLeaderTermsDecoder.term());
    }

    if (bodyDecoder.versionLength() > 0) {
      bodyDecoder.wrapVersion(version);
    } else {
      bodyDecoder.skipVersion();
    }

    assert bodyDecoder.limit() == frameEnd
        : "Decoder read only to position "
            + bodyDecoder.limit()
            + " but expected "
            + frameEnd
            + " as final position";
  }
```